src/lib/coda_networking/coda_networking.ml

```diff
@@ -509,41 +515,42 @@ module Make (Inputs : Inputs_intf) = struct
     loop peers 1
 
   let get_ancestry t preferred_peer input =
-    let input_in_envelope = envelope_from_me t input in
-    (* try preferred_peer first *)
-    let%bind ancestors_or_error =
-      Gossip_net.query_peer t.gossip_net preferred_peer
-        Rpcs.Get_ancestry.dispatch_multi input_in_envelope
-    in
-    let get_random_peers () =
-      let max_peers = 15 in
-      (* 1 + 2 + 4 + 8 *)
-      let except = Peer.Hash_set.of_list [preferred_peer] in
-      random_peers_except t max_peers ~except
-    in
-    match ancestors_or_error with
-    | Ok (Some ancestors) -> return (Ok ancestors)
-    | Ok None ->
-        (* #TODO: punish *)
-        Logger.faulty_peer t.log
-          !"get_ancestry returned no ancestors for the transition sender \
-            %{sexp: Peer.t}, trying non-preferred peers"
-          preferred_peer ;
-        let peers = get_random_peers () in
-        get_ancestry_non_preferred_peers t input_in_envelope peers
-    | Error e ->
-        Logger.warn t.log
-          !"get_ancestry generated error for the transition sender %{sexp: \
-            Peer.t}: %{sexp: Error.t}; trying non-preferred peers"
-          preferred_peer e ;
-        let peers = get_random_peers () in
-        get_ancestry_non_preferred_peers t input_in_envelope peers
+    O1trace.measure "get_ancestry" (fun () ->
```

Comment:
This should probably be `trace_recurring_task` because it has async work happening in it. You should use something other than `measure` if you want to measure the end-to-end latency of an asynchronous operation.